Parse/src/main/java/com/parse/ParseRESTQueryCommand.java

```diff
@@ -41,38 +50,41 @@
     HashMap<String, String> parameters = new HashMap<>();
     List<String> order = state.order();
     if (!order.isEmpty()) {
-      parameters.put("order", ParseTextUtils.join(",", order));
+      parameters.put(KEY_ORDER, ParseTextUtils.join(",", order));
     }
 
     ParseQuery.QueryConstraints conditions = state.constraints();
     if (!conditions.isEmpty()) {
       JSONObject encodedConditions =
           (JSONObject) encoder.encode(conditions);
-      parameters.put("where", encodedConditions.toString());
+      parameters.put(KEY_WHERE, encodedConditions.toString());
     }
 
     // This is nullable since we allow unset selectedKeys as well as no selectedKeys
     Set<String> selectedKeys = state.selectedKeys();
     if (selectedKeys != null) {
-      parameters.put("keys", ParseTextUtils.join(",", selectedKeys));
+      parameters.put(KEY_KEYS, ParseTextUtils.join(",", selectedKeys));
     }
 
     Set<String> includeds = state.includes();
     if (!includeds.isEmpty()) {
-      parameters.put("include", ParseTextUtils.join(",", includeds));
+      parameters.put(KEY_INCLUDE, ParseTextUtils.join(",", includeds));
+    }
+
+    // Respect what the caller wanted for limit, even if count is true, because
+    // parse-server supports it. Currently with our APIs, when counting, limit will always be 0,
+    // but that logic is in ParseQuery class and we should not do that again here.
+    int limit = state.limit();
+    if (limit >= 0) {
+      parameters.put(KEY_LIMIT, Integer.toString(limit));
     }
 
     if (count) {
-      parameters.put("count", Integer.toString(1));
+      parameters.put(KEY_COUNT, Integer.toString(1));
```

Comment:
The default `limit` is `-1` if not set.
https://github.com/parse-community/Parse-SDK-Android/blob/master/Parse/src/main/java/com/parse/ParseQuery.java#L347
So, if there is no `limit` in the query (will be assigned `-1`), then there should be a logic that sets `limit` to `0`?
`parameters.put(KEY_LIMIT, Integer.toString(0));`